### PR TITLE
REF: move putmask internals in array_algos.putmask

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/array_algos/putmask.py [hunk 1]
```diff
@@ -120,3 +120,37 @@ def _putmask_preserve(new_values: np.ndarray, new, mask: np.ndarray):
     except (IndexError, ValueError):
         new_values[mask] = new
     return new_values
+
+
+def putmask_without_repeat(values: np.ndarray, mask: np.ndarray, new: Any) -> None:
```
Comment (jreback) [inline pandas/core/array_algos/putmask.py:125]: might be naming `putmask_with_repeat_inline` (but can change later as well)

Comment (jbrockmendel) [reply pandas/core/array_algos/putmask.py:125]: having it return None should be clear enough

Comment (jreback) [reply pandas/core/array_algos/putmask.py:125]: sure but names matter

Comment (jbrockmendel) [reply pandas/core/array_algos/putmask.py:125]: yah, i just happen to like this name

**Pull request merged**